ctapipe/image/image_processor.py

```diff
@@ -209,51 +211,48 @@ def _parameterize_image(
         # parameterization
         return default
 
-    def _process_telescope_event(self, event):
+    def _process_telescope_event(self, tel_event: TelescopeEventContainer):
         """
         Loop over telescopes and process the calibrated images into parameters
         """
-        for tel_id, dl1_camera in event.dl1.tel.items():
+        tel_id = tel_event.index.tel_id
+        dl1 = tel_event.dl1
 
-            if self.apply_image_modifier.tel[tel_id]:
-                dl1_camera.image = self.modify(tel_id=tel_id, image=dl1_camera.image)
+        if self.apply_image_modifier.tel[tel_id]:
+            dl1.image = self.modify(tel_id=tel_id, image=dl1.image)
 
-            dl1_camera.image_mask = self.clean(
-                tel_id=tel_id,
-                image=dl1_camera.image,
-                arrival_times=dl1_camera.peak_time,
-            )
+        dl1.image_mask = self.clean(
+            tel_id=tel_id,
+            image=dl1.image,
+            arrival_times=dl1.peak_time,
+        )
+        dl1.parameters = self._parameterize_image(
+            tel_id=tel_id,
+            image=dl1.image,
+            signal_pixels=dl1.image_mask,
+            peak_time=dl1.peak_time,
+            default=self.default_image_container,
+        )
 
-            dl1_camera.parameters = self._parameterize_image(
-                tel_id=tel_id,
-                image=dl1_camera.image,
-                signal_pixels=dl1_camera.image_mask,
-                peak_time=dl1_camera.peak_time,
-                default=self.default_image_container,
+        self.log.debug("params: %s", dl1.parameters.as_dict(recursive=True))
+
+        if (
+            tel_event.simulation is not None
+            and tel_event.simulation.true_image is not None
+        ):
+            simulation = tel_event.simulation
+            simulation.true_parameters = self._parameterize_image(
+                tel_id,
+                image=simulation.true_image,
+                signal_pixels=simulation.true_image > 0,
+                peak_time=None,  # true image from simulation has no peak time
+                default=DEFAULT_TRUE_IMAGE_PARAMETERS,
             )
+            for container in simulation.true_parameters.values():
```

Comment:
If the prefix doesn't start with "true_", doesn't that mean there's something wrong with the file and we should raise an error rather than silently correcting it?

Reply:
> If the prefix doesn't start with "true_", doesn't that mean there's something wrong with the file 

Has nothing to do with the input file, since the container is created by the `_parametrize_image` methods, but you are right, this is confusing.

I will check if it is needed / can be improved.